lib/matplotlib/widgets.py

```diff
@@ -777,6 +784,38 @@ def __init__(
         self._active_handle = None
         self.set_val(valinit)
 
+    def _fill_verts(self, verts, val):
+        """
+        Update the *verts* of the *self.poly* slider according to the current
+        value and slider orientation.
+
+       Parameters
+       ----------
+       verts : (5, 2) np.ndarray
+            The vertices to fill. Note that these will be modified in
+            place.
+       val : (2,) array_like
+            The new min and max values.
+
+       Returns
+       -------
+       verts : np.ndarray
+            The original array with updated values.
+        """
+        if self.orientation == "vertical":
+            verts[0] = .25, val[0]
+            verts[1] = .25, val[1]
+            verts[2] = .75, val[1]
+            verts[3] = .75, val[0]
+            verts[4] = .25, val[0]
+        else:
+            verts[0] = val[0], .25
+            verts[1] = val[0], .75
+            verts[2] = val[1], .75
+            verts[3] = val[1], .25
+            verts[4] = val[0], .25
+        return verts
```

Comment:
I'd pull the interface logic one level up. - No need to bother with vertices in the function call:

```suggestion
    def _update_selection_poly(self, vmin, vmax):
        """
        Update the vertices of the *self.poly* slider in-place
        to cover the data range *vmin*, *vmax*.
        """
        # The vertices are positioned
        #  1 ------ 2
        #  |        |
        # 0, 4 ---- 3 
        verts = self.poly.xy
        if self.orientation == "vertical":
            verts[0] = verts[4] = .25, vmin
            verts[1] = .25, vmax
            verts[2] = .75, vmax
            verts[3] = .75, vmin
        else:
            verts[0] = verts[4] = vmin, .25
            verts[1] = vmin, .75
            verts[2] = vmax, .75
            verts[3] = vmax, .25
```